tools_docker/test_docker/README.md

```diff
@@ -0,0 +1,45 @@
+# Docker Tests
+
+This is a tool to help test Docker images, for simple cases, but also for complicated tests.
+This allows:
+- automatically running Tracim with Collabora and Elasticsearch on by default (it just needs some tweaks
+for emails and you need to setup dns resolving using a special Docker image)
+- testing if the Docker container correctly starts
+- automatically testing services running in the docker container
+
+## Prerequisities
+
+- install some python packages:
+`pip install pytest testinfra python-dotenv`
+- build tracim docker image `DOCKER_TRACIM_IMAGE` (see test_config.env)
+- build elasticsearch docker image `DOCKER_ELASTICSEARCH_IMAGE`
+- create tracim env config file from sample:`cp tracim.sample.env tracim.env`
+- create docker test config file from sample: `cp test_config.sample.env test_config.env`
+
+for hostname resolving from host, use:
+
+```sh
+docker run --name dns_docker_solver --hostname dns.mageddo --restart=unless-stopped -p 5380:5380 \
+-v /var/run/docker.sock:/var/run/docker.sock \
+-v /etc/resolv.conf:/etc/resolv.conf \
```

Comment:
- `--restart=unless-stopped` is it necessary for testing ?
- do you know if some impact exist on host machine (example if you want to no more use dns_docker_solver
~~~
-v /var/run/docker.sock:/var/run/docker.sock \
-v /etc/resolv.conf:/etc/resolv.conf \
~~~

Reply:
>   --restart=unless-stopped is it necessary for testing ?
Not necessary in fact. we can run this with different mecanism, it's just a bit more practical to test with a configuration that doesn't force developer to always rethink about running this or this docker.

> do you know if some impact exist on host machine (example if you want to no more use dns_docker_solver)
- This docker make docker with hostname accessible locally by these hostname, so it can impact other dockers and host machine this way.
- I don't find any unwanted impact, but it effectively change a bit the local machine dns process (this is the goal of this docker)

Note: This is a really useful trick to use domain name in a local environment for docker image. I don't remembers exactly but i do need this to do proper matching between collabora and tracim domain (we do need to setup collabora url from tracim and without this, it's really hard to guess it).
see more info about this here: https://hub.docker.com/r/defreitas/dns-proxy-server